src/main/java/DukeException.java

```diff
@@ -0,0 +1,14 @@
+/**
+ * <p>The DukeException class defines the behavior of an exception that is produced by Duke.</p>
+ */
+public class DukeException extends Exception {
+    protected final String message;
+    public DukeException(String message) {
```

Comment:
The base class `java::lang::Exception` already has a constructor taking a message, so i think it would be less repetitive to use that, so you can do `super(message)` instead?